### PR TITLE
Fix log send crash. Fix attributes missing for keybase folder

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] The logs need to be under a folder and not the root in the homedir. This works on the sim but not on the device
- [x] We need to set flags to stop the keybase folder from being uploaded to icloud. I think the best way to verify this is to look into the sim folder data direction and run xattr. I see
```
xattr *
Keybase: com.apple.metadata:com_apple_backup_excludeItem
```  
which is correct

@keybase/react-hackers

### CODE REVIEW

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 1]
```diff
@@ -22,6 +22,20 @@ @interface AppDelegate ()
 
 @implementation AppDelegate
 
+- (BOOL)addSkipBackupAttributeToItemAtPath:(NSString *) filePathString
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: from https://developer.apple.com/library/content/qa/qa1719/_index.html

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 2]
```diff
@@ -22,6 +22,20 @@ @interface AppDelegate ()
...
+{
+  NSURL* URL= [NSURL fileURLWithPath: filePathString];
+  assert([[NSFileManager defaultManager] fileExistsAtPath: [URL path]]);
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: Don't use asserts... if you really want to use NSAssert in objc

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 3]
```diff
@@ -22,6 +22,20 @@ @interface AppDelegate ()
...
+
+  NSError *error = nil;
+  BOOL success = [URL setResourceValue: [NSNumber numberWithBool: YES]
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: [URL setResourceValue:@(YES)]

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 4]
```diff
@@ -22,6 +22,20 @@ @interface AppDelegate ()
 
 @implementation AppDelegate
 
+- (BOOL)addSkipBackupAttributeToItemAtPath:(NSString *) filePathString
+{
+  NSURL* URL= [NSURL fileURLWithPath: filePathString];
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: If you want, try to make the obj consistent with rest of file? There are extra spaces for args.

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 5]
```diff
@@ -36,7 +50,18 @@ - (void) setupGo
 
 #if TESTING
 #else
-  NSString * logFile = [home stringByAppendingPathComponent:@"ios.log"];
+  NSString * library = [home stringByAppendingPathComponent:@"Library"];
+  NSString * appSupport = [library stringByAppendingPathComponent:@"Application Support"];
+  NSString * keybasePath = [appSupport stringByAppendingPathComponent:@"Keybase"];
+  NSString * logFile = [keybasePath stringByAppendingPathComponent:@"ios.log"];
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: You can use slashes?

NSString * keybasePath = @"~/Library/Application Support/Keybase";

**Comment by gabriel** [comment]
Oh use ~/Library/Caches for logs. It's not backed up.

File: shared/react-native/ios/Keybase/AppDelegate.m (hunk 2 above)
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: i just pulled this from the apple code, didn't want to mutate it much

Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: but i can make the style changes, i don't care

Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: Apple sample code is notoriously bad.

You should definitely remove the assert though.

File: shared/react-native/ios/Keybase/AppDelegate.m (hunk 5 above)
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: k, i'll use stringByExpandingTildeInPath so we continue to send abs paths to go

**Comment by chrisnojima** [comment]
@gabriel updated

**Comment by chrisnojima** [comment]
@gabriel / @keybase/react-hackers please review

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 6]
```diff
@@ -36,7 +47,16 @@ - (void) setupGo
 
 #if TESTING
 #else
-  NSString * logFile = [home stringByAppendingPathComponent:@"ios.log"];
+  NSString * keybasePath = [@"~/Library/ Application Support/Keybase" stringByExpandingTildeInPath];
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: @"~/Library/ Application Support/Keybase" => @"~/Library/Application Support/Keybase"

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 7]
```diff
@@ -36,7 +47,16 @@ - (void) setupGo
...
+  NSString * logFile = [@"~/Library/Caches/ios.log" stringByExpandingTildeInPath];
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/AppDelegate.m]: `"~/Library/Caches/Keybase/ios.log"`

**Comment by chrisnojima** [comment]
@gabriel fixed

**Pull request merged**